application.go

```diff
@@ -973,3 +890,45 @@ func (d *Discovery) AddPort(port Port) *Discovery {
 	d.Ports = &ports
 	return d
 }
+
+// EmptyNetworks explicitly empties networks
+func (r *Application) EmptyNetworks() *Application {
+	r.Networks = &[]Network{}
+	return r
+}
+
+// BridgedNetwork sets the networking mode to bridged
+func (r *Application) BridgedNetwork(name string) *Application {
+	if r.Networks == nil {
+		r.EmptyNetworks()
+	}
+	network := Network{Name: name, Mode: "container/bridge"}
+	networks := *r.Networks
+	networks = append(networks, network)
+	r.Networks = &networks
+	return r
+}
+
+// ContainerNetwork sets the networking mode to container
+func (r *Application) ContainerNetwork(name string) *Application {
+	if r.Networks == nil {
+		r.EmptyNetworks()
+	}
+	network := Network{Name: name, Mode: "container"}
+	networks := *r.Networks
+	networks = append(networks, network)
+	r.Networks = &networks
+	return r
+}
+
+// HostNetwork sets the networking mode to host
+func (r *Application) HostNetwork(name string) *Application {
```

Comment:
Instead of having three different methods to set the network mode, let's have a single one that takes a custom string type denoting the possible modes (which we can provide from a set of constants).